[WIP]org settings: Fix the spacing/indentation of dependent input elements.

shubhamdhama opened the pull request:
First followup of https://github.com/zulip/zulip/pull/9071#issuecomment-384465530
![dependent](https://user-images.githubusercontent.com/22238472/39399212-88889f74-4b37-11e8-84ae-391f4ca1c41b.png)
![depend 2](https://user-images.githubusercontent.com/22238472/39399213-88df23da-4b37-11e8-9a70-cec3679f7bf8.png)
Feedback needed @rishig  @timabbott

rishig commented:
cool! For the indent, I think we should either align it with the left edge, or indent it a bit further.

shubhamdhama commented:
Here are some other tries:
30 px: ![30px](https://user-images.githubusercontent.com/22238472/39399540-3b9926a6-4b3d-11e8-8870-3a98dcbcc558.png)
20 px: ![20px](https://user-images.githubusercontent.com/22238472/39399541-3bdcc0dc-4b3d-11e8-82b6-99812b7e8816.png)
0 px: ![0px](https://user-images.githubusercontent.com/22238472/39399542-3c397f2a-4b3d-11e8-9de9-02bf2789fd91.png)
@rishig FYI.

shubhamdhama commented:
Changed the look of notification setting buttons, why one of these looks better:

40 px: ![buttons](https://user-images.githubusercontent.com/22238472/39400432-f0479e1a-4b4d-11e8-94f8-63d5bd43ae33.png)
------
20px: ![20pxnotifbuton](https://user-images.githubusercontent.com/22238472/39400434-f0eb9948-4b4d-11e8-9f59-64c7a42fd498.png)
------
0px: ![0pxnotif](https://user-images.githubusercontent.com/22238472/39400435-f1359c64-4b4d-11e8-86c5-55d29f989107.png)
-------

shubhamdhama commented:
Second followup of https://github.com/zulip/zulip/pull/9071#issuecomment-384465530
![domaintable](https://user-images.githubusercontent.com/22238472/39400450-411cc748-4b4e-11e8-8d3c-42e4a0cd7b70.gif)

timabbott commented:
For the notification buttons, we should use the button CSS styles we use elsewhere in Zulip, not the default bootstrap green buttons.

shubhamdhama commented:
![newbuttons](https://user-images.githubusercontent.com/22238472/39406068-4f874efe-4bce-11e8-9fba-aae10df963e5.png)
Here is how button looks with `sea-green` class.

timabbott commented:
Yeah, I think that button look is reasonable; 40px looks best to me of your options; I think we basically want the label to be aligned with the label for e.g. "Visual desktop notifications".  

For the dropdowns, 30px looks best to me; @rishig thoughts?

timabbott commented:
Hmm, yeah, agreed on notification settings.

rishig commented:
Or possibly we should just embrace the `N`, and do
`[Up to N minutes after posting]  N: [input box]`

With the N on the same line as the drop down.

shubhamdhama commented:
@rishig @timabbott I've updated the PR:
* dropdown with the left margin 35px(also removed 100% unnecessary width from messaged edit drop-down) 
![dropdown](https://user-images.githubusercontent.com/22238472/39473770-1de1716c-4d6e-11e8-8c1b-0055d05e949d.png)
![dropdonw2](https://user-images.githubusercontent.com/22238472/39473771-1e23c940-4d6e-11e8-97a4-6b4c5efd04da.png)


* For notification buttons, I've reverted them to previous bootstrap button used "bot edit" buttons.
![notfbuton](https://user-images.githubusercontent.com/22238472/39473774-2199f612-4d6e-11e8-9c64-8ab6c81d9817.png)


I'm still unsure about the feedback of this comment https://github.com/zulip/zulip/pull/9257#issuecomment-385493076 , if the consensus is to go with this approach I can give this a try.

rishig commented:
Chatted with Tim, let's indeed go with the approach at https://github.com/zulip/zulip/pull/9257#issuecomment-385493076.

What I meant is that the dropdown option would be: "Up to N minutes after posting"
And then if you select that option, to the right of the dropdown, it would appear as
`N: [   ]`

Does that make sense?

As followup we can change the styling for the similar option in Organization permissions -> Who can create streams.

shubhamdhama commented:
Here is how it looks now:

![better dropdown](https://user-images.githubusercontent.com/22238472/39538219-b4ded30c-4e59-11e8-9adc-6695b55d9dd8.png)
I also liked this personally!

shubhamdhama commented:
![screenshot from 2018-05-02 23-13-32](https://user-images.githubusercontent.com/22238472/39539844-7dbddb84-4e5e-11e8-90ad-c7c51bec24c2.png)
@rishig  How does this look?
Aligned N to the bottom and decreased top-bottom padding to 2px(Or 1px is better here)?

timabbott commented:
I think 1px is better, but maybe we actually just want to use something to get vertical centering rather than playing with pixels.

rishig commented:
I deleted the comment, but I meant smaller horizontally, for the input box. But on second thought I think it's actually fine as it was; someone might want to enter 2880 for 2 days, for instance.

The second N should be aligned with the text baseline, e.g. with the N in the dropdown.

shubhamdhama commented:
@rishig @timabbott Updated here's how it looks now(only changed the vertical position of 'N'):
![screenshot from 2018-05-03 12-25-20](https://user-images.githubusercontent.com/22238472/39563280-16b464fc-4ecd-11e8-8f2a-963cd5f4c430.png)

timabbott commented:
That looks good to me.

rishig commented:
looks great!

timabbott commented:
I think the quotes around `N` are a regression; instead we should just add `N` to the exclude list in `tools/lib/capitalization.py`

Otherwise lgtm.

timabbott commented:
@shubhamdhama BTW, nobody gets notified when you push a new version, so you need to comment when you're ready for another review; I think that's how this one got lost :)

shubhamdhama commented:
Resolved the conflicts(waiting for the test to get passed)
>nobody gets notified when you push a new version, so you need to comment when you're ready for another review; I think that's how this one got lost :)

Ohh, yes, I realized that I forgot to comment here after telling the update at czo. My bad 😓
